packages/components/menu/src/menu.ts

```diff
@@ -180,7 +180,9 @@ export default defineComponent({
       if (isOpened) {
         closeMenu(index, indexPath)
       } else {
-        openMenu(index, indexPath)
+        activeIndex.value?.startsWith(index)
```

Comment:
Thank for your contribution, but the `index` is user-defined and we cannot judge it this way